### ISSUE TITLE
Apply ocean depth fixes for cnrm_esm2_1, gfdl_esm4, ipsl_cm6a_lr 

### ISSUE THREAD

**Issue by tomaslovato** [ESMValGroup/ESMValCore#1069]
This activity relates to the development of ESMValTool recipes from CE2COAST project (https://github.com/ESMValGroup/ESMValTool/issues/2114) that require a number of fixes in ESMValCore.
The recipe(s) from CE2COAST project deal primarily with **oceanic parameters**, e.g., thetao, no3, po4 ...

1. There is the need to **propagate the time coordinate also after the climate statistics** to correctly create a sequence of preprocessing operations (see details in issue https://github.com/ESMValGroup/ESMValCore/issues/1018)

2. A number of datasets with incorrect definition of ocean depth coordinate in `Omon` files:
- **cnrm_esm2_1, gfdl_esm4, ipsl_cm6a_lr** : have wrong standard name of level coordinate
- **cesm2**: have wrong standard name of level coordinate and depth coordinate is in `centimeters` instead of `meters`, but not in level bounds (see also https://github.com/ESMValGroup/ESMValCore/issues/859 and https://github.com/ESMValGroup/ESMValCore/pull/888)

As the issue with depth coordinate affect most of 3D variables in `Omon`, it would be ideal to create a fix that operate at the mip level, instead of declaring a fix for each variable. 
@valeriupredoi do you think this could be a viable solution?

**Comment by valeriupredoi** [comment]
assigning @ledm because he is Mr Ocean and @zklaus because the data needs core fixes(possibly at MIP level). I personally don't think this will be possible at MIP level because that involves regenerating the data and re-uploading on ESGF (which is a royal pain in the bahoot). I am in favour of preserving/creating the time axis in preprocessor - what do others think?

**Comment by tomaslovato** [comment]
@valeriupredoi @zklaus I think I could have misused the term `mip`, as I was referring to the variable name used in [`get_fixes`](https://github.com/ESMValGroup/ESMValCore/blob/9f4553ece7fc731fd3bf1b9fed2d7d6b87216e61/esmvalcore/cmor/_fixes/fix.py#L124-L172) function of `fix.py`
Actually, my solution was to introduce in the check of possible `fix_name` also the `mip` category that refers to the cmor variable Tables (e.g. `Omon`, `Oyr` etc )  as in the following code 
``` diff
             classes = inspect.getmembers(fixes_module, inspect.isclass)
             classes = dict((name.lower(), value) for name, value in classes)
-            for fix_name in (short_name, 'allvars'):
+            for fix_name in (short_name, mip.lower(), 'allvars'):
                 try:
                     fixes.append(classes[fix_name](vardef))
                 except KeyError:
```
so we can create a fix class that applies using the individual variable `short_name`, a cmor Table (e.g.`Omon`), or the entire dataset with `allvars`.
The only drawback I see is that if additional variable fixes are required (e.g., scaling, unit correction) within the `Omon` table, it should be added within this "mip" class.
Probably it is faster to see it coded ...

**Comment by valeriupredoi** [comment]
ah got you! @jvegasbsc what do you think? I reckon we could do this relatively easy, only caveat I can think of is that there may be models that actually have output that doesn't need to be fixed in a certain mip, so we should be careful not to fix data that's not broken

**Comment by tomaslovato** [comment]
@valeriupredoi @jvegasbsc 
Actually the fix for the new `mip` class applies only to the problematic dataset, not to all... have a look to the proposed changes for `cesm2` dataset fixes in 1464c52be
This is simply an extension of the possible `Classes` that we can define to create a fix ...

V-man, I also added a fix to the depth coordinate of `cesm2` data that should address #888, as the same problem is in there for other ocean 3D parameters (e.g, no3, po4 ... )

**Comment by tomaslovato** [comment]
As discussed in https://github.com/ESMValGroup/ESMValCore/pull/1095#issuecomment-832043564 this issue has been readdressed to apply fixes for the ocean depth coordinate of cnrm_esm2_1, gfdl_esm4, ipsl_cm6a_lr datasets, once that #1095 is merged
